android/CouchbaseLite/src/main/java/com/couchbase/lite/ConsoleLogger.java

```diff
@@ -0,0 +1,96 @@
+//
+// ConsoleLogger.java
+//
+// Copyright (c) 2018 Couchbase, Inc All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package com.couchbase.lite;
+
+import android.util.Log;
+
+import com.couchbase.litecore.C4Log;
+
+import java.util.ArrayList;
+
+public final class ConsoleLogger implements Logger {
+    private LogLevel _level = LogLevel.WARNING;
+    private ArrayList<LogDomain> _domains = new ArrayList<>();
+
+    ConsoleLogger() {
+        _domains.add(LogDomain.ALL);
+    }
+
+    public ArrayList<LogDomain> getDomains() {
+        return _domains;
+    }
+
+    public void setDomains(ArrayList<LogDomain> domains) {
+        _domains = domains;
+    }
+
+    public void setLevel(LogLevel level) {
+        if(level == null) {
```

Comment:
Maybe we should guard null LogLevel? or null would mean `NONE`?

How does .NET handle this?

Reply:
So let to NONE if null is set?

Reply:
It's up to you, this is a situation specific to the Java language.

Reply:
Then let's do NONE.